tools/cldr-code/src/main/java/org/unicode/cldr/util/LocalesTxtReader.java

```diff
@@ -0,0 +1,205 @@
+                } catch (Exception e) {
+                    throw new IllegalArgumentException("Invalid organization in Locales.txt: " + line);
+                }
+
+                // verify that the locale is valid BCP47
+                String localePart = stuff.get(1).trim();
+                List<String> localeStuff = CldrUtility.splitList(localePart, ' ', true);
+                Set<String> locales = new TreeSet<>();
+
+                for (final String entry : localeStuff) {
+                    if (entry.startsWith("@")) {
+                        List<String> kwStuff = CldrUtility.splitList(entry, '=', true);
+                        if (kwStuff.size() > 2 || kwStuff.size() < 1) {
+                            throw new IllegalArgumentException("Invalid @-command " + entry + " in Locales.txt: " + line);
+                        }
+                        final String atCommand = kwStuff.get(0);
+                        switch(atCommand) {
+                            case "@weight":
+                                weight = Integer.parseInt(kwStuff.get(1));
+                                break;
+
+                            case "@pathMatch":
+                                pathMatch = kwStuff.get(1);
+                            break;
+                            default:
+                                throw new IllegalArgumentException("Unknown @-command " + atCommand + " in Locales.txt: " + line);
+                        }
+                    } else {
+                        locales.add(entry);
+                    }
+                }
+
+                if (locales.size() != 1) {
+                    // require there to be exactly one locale.
+                    // This would allow collapsing into fewer lines.
+                    throw new IllegalArgumentException("Expected one locale entry in Locales.txt but got " + locales.size() + ": " + line);
+                }
+
+                // extract the single locale, process as before
+                String locale = locales.toArray(new String[1])[0];
+
+                if (!locale.equals(StandardCodes.ALL_LOCALES)) {
+                    parser.set(locale);
+                    String valid = sc.validate(parser);
+                    if (valid.length() != 0) {
+                        throw new IllegalArgumentException("Invalid locale in Locales.txt: " + line);
+                    }
+                    locale = parser.toString(); // normalize
+
+                    // verify that the locale is not a default content locale
+                    if (defaultContentLocales.contains(locale)) {
+                        throw new IllegalArgumentException("Cannot have default content locale in Locales.txt: " + line);
+                    }
+                }
+
+                Level status = Level.get(stuff.get(2));
+                if (status == Level.UNDETERMINED) {
+                    System.out.println("Warning: Level unknown on: " + line);
```

Comment:
Let's make this an exception and fail. We don't want stuff be be screwed in up production.

Reply:
Sounds like a good improvement. Note that this function was moved from StandardCodes.java